src/envoy/mixer/grpc_transport.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "src/envoy/mixer/grpc_transport.h"
+
+#include "common/grpc/rpc_channel_impl.h"
+
+using ::google::protobuf::util::Status;
+using StatusCode = ::google::protobuf::util::error::Code;
+
+namespace Envoy {
+namespace Http {
+namespace Mixer {
+namespace {
+
+// gRPC request timeout
+const std::chrono::milliseconds kGrpcRequestTimeoutMs(5000);
```

Comment:
I did not read Envoy code,  I guess it will be min(this_timeout, and cluster_time_out) for each request